doc/email-notificaitons.md

```diff
@@ -0,0 +1,58 @@
+# Email notifications
+The application sends various notifications by email, this document outlines how
+this works from a technical standpoint.
+
+## Account emails
+Most emails that relate to user accounts are sent by Auth0.
+
+All others are sent by the applicaiton via GOV.UK Notify.
+
+## Mail Notify Gem
+We use this gem to interact with the GOV.UK Notify service.
+
+[https://github.com/dxw/mail-notify](https://github.com/dxw/mail-notify)
+
+## GOV.UK notify service
+[https://www.notifications.service.gov.uk/services/15322373-3ee0-4c9c-aef7-be3e967f2f3e](https://www.notifications.service.gov.uk/services/15322373-3ee0-4c9c-aef7-be3e967f2f3e)
+
+BEIS owned so access can be granted by them.
+
+You will need the API key in the NOTIFY_KEY environment variable. The key for
+each environment can be found here https://www.notifications.service.gov.uk/services/15322373-3ee0-4c9c-aef7-be3e967f2f3e/api/keys
+
+## Templates
+We use two templates to send email notifications:
+
+- Welcome email
+- View email
+
+See the templates at https://www.notifications.service.gov.uk/services/15322373-3ee0-4c9c-aef7-be3e967f2f3e/templates
+
```

Comment:
is it ok to expose this ID in the repo?

Reply:
I think so, my rational is:

- it is behind a secure login
- the id is for the service account and could just as easily be 'beis-roda'
- linking to the exact templates benefits the user, they can be confident they have the right account and it would idicate if they do not have access to the right account

Happy to take it out if you are converned though?

Reply:
I think this is fine. We have the UUIDs exposed in the example `.env` files. There's nothing any potential attacker can do with these.